packages/core/src/types.ts

```diff
@@ -1960,3 +2008,13 @@ export type StateValueFrom<TMachine extends AnyStateMachine> = Parameters<
 
 export type StateFromMachine<TMachine extends AnyStateMachine> =
   TMachine['initialState'];
+
+export type PersistedMachineState<TState extends AnyState> = TState & {
```

Comment:
This intersection is definitely lying to us :P it should look closer to this one:
```suggestion
export type PersistedMachineState<TState extends AnyState> = Pick<
  TState,
  'value' | 'output' | 'context' | '_event' | 'done' | 'historyValue'
> & {
```